dags/python_scripts/atd_vzd_cr3_scan_pdf_records.py

```diff
@@ -0,0 +1,300 @@
+    )
+    response.encoding = "utf-8"
+    try:
+        return response.json()["data"]["atd_txdot_crashes"][0]["cr3_file_metadata"]
+    except (IndexError, TypeError, KeyError):
+        return None
+
+
+def get_records(limit: int = 100) -> Set[int]:
+    """
+    Returns a list of all CR3 crashes without CR3 PDF metadata
+    :return Set[int]:
+    """
+    if not str(limit).isdigit() or limit == 0:
+        return set()
+
+    # Executes GraphQL to fetch any crashes without CR3 PDF metadata, with a limit as indicated
+    query_get_records = """
+        query getUnverifiedCr3($limit:Int!) {
+          atd_txdot_crashes(where: {cr3_file_metadata: {_is_null: true}, cr3_stored_flag: {_eq: "Y"}}, limit: $limit, order_by: {crash_id: desc}) {
+            crash_id
+          }
+        }
+    """
+    response = requests.post(
+        url=os.getenv("HASURA_ENDPOINT"),
+        headers={
+            "Accept": "*/*",
+            "content-type": "application/json",
+            "x-hasura-admin-secret": os.getenv("HASURA_ADMIN_KEY")
+        },
+        json={
+            "query": query_get_records,
+            "variables": {
+                "limit": int(limit)
+            }
+        }
+    )
+    response.encoding = "utf-8"
+    records = response.json()
+
+    return set(map(lambda x: x["crash_id"], records["data"]["atd_txdot_crashes"]))
+
+
+def update_metadata(crash_id: int, metadata: dict) -> bool:
+    """
+    Returns True if it manages to update the metadata for an object.
+    :param int crash_id: The crash id to be updated
+    :param dict metadata: The metadata to be set
+    :return bool: 
+    """
+    if not is_crash_id(crash_id) \
+            or not isinstance(metadata, dict) \
+            or not is_valid_metadata(metadata):
+        return False
+
+    print(f"Updating crash_id: {crash_id}, with metadata: {json.dumps(metadata)}")
+
+    mutation_update_crash_cr3_metadata = """
+        mutation updateCrashCr3Metadata($crash_id:Int!, $metadata: jsonb) {
+          update_atd_txdot_crashes(where:{crash_id:{_eq: $crash_id}}, _set: {cr3_file_metadata: $metadata}){
+            affected_rows
+          }
+        }
+    """
+    response = requests.post(
+        url=os.getenv("HASURA_ENDPOINT"),
+        headers={
+            "Accept": "*/*",
+            "content-type": "application/json",
+            "x-hasura-admin-secret": os.getenv("HASURA_ADMIN_KEY")
+        },
+        json={
+            "query": mutation_update_crash_cr3_metadata,
+            "variables": {
+                "crash_id": int(crash_id),
+                "metadata": metadata
+            }
+        }
+    )
+    response.encoding = "utf-8"
+    try:
+        return response.json()["data"]["update_atd_txdot_crashes"]["affected_rows"] > 0
+    except (KeyError, TypeError):
+        return False
+
+
+def main():
+    """
+    Main Loop
+    """
+    # Processes each one of these crashes using process_record function
+    for crash_id in get_records(10):
```

Comment:
did you mean to leave the `limit` at 10 here?

Reply:
Ah, yikes. That should have been an environment variable. Thanks for catching that. 🙏